chips/nrf52/src/ieee802154_radio.rs

```diff
@@ -767,128 +797,250 @@ impl<'a> Radio<'a> {
         buffer
     }
 
-    // TODO: Theres an additional step for 802154 rx/tx handling
+    fn crc_check(&self) -> Result<(), ErrorCode> {
+        if self.registers.crcstatus.is_set(Event::READY) {
+            Ok(())
+        } else {
+            Err(ErrorCode::FAIL)
+        }
+    }
+
+    // TODO: RECEIVING ACK FOR A SENT TX IS NOT IMPLEMENTED //
+    // As a general note for the interrupt handler, event registers must still be cleared even when
+    // hardware shortcuts are enabled.
     #[inline(never)]
     pub fn handle_interrupt(&self) {
         self.disable_all_interrupts();
 
-        if self.registers.event_ready.is_set(Event::READY) {
-            self.registers.event_ready.write(Event::READY::CLEAR);
-            self.registers.event_end.write(Event::READY::CLEAR);
-            if self.transmitting.get()
-                && self.registers.state.get() == nrf5x::constants::RADIO_STATE_RXIDLE
-            {
-                self.registers.task_ccastart.write(Task::ENABLE::SET);
-            } else {
-                self.registers.task_start.write(Task::ENABLE::SET);
-            }
-        }
+        let mut start_task = false;
+        let mut rx_init = false;
 
-        if self.registers.event_framestart.is_set(Event::READY) {
-            self.registers.event_framestart.write(Event::READY::CLEAR);
-        }
-
-        //   IF we receive the go ahead (channel is clear)
-        // THEN start the transmit part of the radio
-        if self.registers.event_ccaidle.is_set(Event::READY) {
-            self.registers.event_ccaidle.write(Event::READY::CLEAR);
-            self.registers.task_txen.write(Task::ENABLE::SET)
-        }
+        match self.state.get() {
+            // It should not be possible to receive an interrupt while the tracked radio state is OFF
+            RadioState::OFF => panic!("Received interrupt while off"),
+            RadioState::RX => {
+                // Enforce READY_START shortcut enablement if radio is receiving
+                if !self.registers.shorts.is_set(Shortcut::READY_START) {
+                    panic!("READY_START shortcut not enabled while sending.")
```

Comment:
This seems like a waste. It's an invariant that's enforced everywhere, and if there is a bug in the logic, we probably don't want it bringing down the whole system. Seems better to just _test_ for it, rather than continuously test at runtime.

Reply:
I agree that this makes more sense. I will go ahead and remove this. Do we have an established methodology or way to test this in future PRs or changes to this driver?